### PR TITLE
[mlir][nvvm] Fix the verifier of `wgmma.mma_async` wrt transposed layouts 

### PR DESCRIPTION
**Pull request by shubaoyu2**
the WGMMA expect layouts for A/B are row/col, the transposed version should be col/row. when checking other datatypes cannot use transposed layout, it should reject col-major for A and row-major for B

### CODE REVIEW

File: mlir/lib/Dialect/LLVMIR/IR/NVVMDialect.cpp [hunk 1]
```diff
@@ -880,7 +880,7 @@ LogicalResult NVVM::WgmmaMmaAsyncOp::verify() {
   // Check transpose (only available for f16/bf16)
```
Comment (grypp) [inline mlir/lib/Dialect/LLVMIR/IR/NVVMDialect.cpp:880]: Can we add a comment here that mentions the required layouts. We can use ehat PTX guideline says

**Review by grypp — COMMENTED** [review]
Looks good. Let's add comments in the code. 

For the title, we use [mlir][nvvm] prefix

File: mlir/lib/Dialect/LLVMIR/IR/NVVMDialect.cpp (hunk 1 above)
Comment (shubaoyu2) [reply mlir/lib/Dialect/LLVMIR/IR/NVVMDialect.cpp:880]: ok, added some description according to PTX guideline

**Review by grypp — APPROVED** [review]
Awesome thanks for working on this

**Comment by grypp** [comment]
I will merge it once the buildkite is done

**Comment by grypp** [comment]
@shubaoyu2 you need to run `clang-format` on your code. The buildkite fails here

**Pull request merged**